core/constraint-streams/src/main/java/ai/timefold/solver/constraint/streams/bavet/bi/BavetBiConcatNode.java

```diff
@@ -16,4 +16,10 @@ public final class BavetBiConcatNode<A, B> extends AbstractConcatNode<BiTuple<A,
     protected BiTuple<A, B> getOutTuple(BiTuple<A, B> inTuple) {
         return new BiTuple<>(inTuple.factA, inTuple.factB, outputStoreSize);
     }
+
+    @Override
+    protected void updateOutTuple(BiTuple<A, B> inTuple, BiTuple<A, B> outTuple) {
+        outTuple.factA = inTuple.factA;
```

Comment:
Tuples have `updateInDifferent(...)` just for this case.
Turns out it is a common operation, so it was extracted already.